### DISCUSSION TITLE
Does updating state always update DOM?

### DISCUSSION THREAD

**Discussion by andywilsonuk** [jorgebucaran/hyperapp#1061] [Q&A] [answered]
Hello!

I've been picking through the hyperapp code and I can't decide whether setting a property to the same value within the state will update the DOM. For example:

```javascript
function action(state) {
  return {
    ...state,
    show: true
  }
}
```

Whereby state.show was already set to true.

I think this will cause the view to regenerate and may or may not update the vdom which may or may not update the dom. I could do a check before the return to see if the value is the same and return the previous state however this simple example is masking some of the complexities of what I'm trying to achieve.

Please could someone clarify what happens when property values are set to the same value within updated state.

Thanks

**Comment by jorgebucaran** [comment, marked as answer]
Updating the state will always update the DOM, but setting a property to the same value **will not** because that does not constitute a state update. We skip DOM updates [here](https://github.com/jorgebucaran/hyperapp/blob/14cc6c4ba64c83703048756fb1065da7ec3a59a1/index.js#L381).

**Reply by andywilsonuk** [reply]
Ah I see, so despite the fact that I'm returning a new literal which causes a new vdom to be generated from the view, upon comparing old and new vdom nodes and properties, the net result is "no change to DOM".

**Reply by jorgebucaran** [reply]
Yes, sorry if I was unclear. Hyperapp will invoke your view to generate new VDOM nodes, but the DOM will not be updated, since there's nothing to update.

If your goal is to go further and avoid creating new VDOM nodes, there's also [`memo()`](https://github.com/jorgebucaran/hyperapp/blob/main/docs/api/memo.md).
